src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -255,10 +257,23 @@ export function getInputType(
 
     function getInputModelType(m: Model): InputType {
         const intrinsicName = getIntrinsicModelName(program, m);
-        if (intrinsicName && intrinsicName === "string") {
-            const values = getKnownValues(program, m);
-            if (values) {
-                return getInputModelForExtensibleEnum(m, values);
+        if (intrinsicName) {
+            switch (intrinsicName) {
+                case "string":
+                    const values = getKnownValues(program, m);
+                    if (values) {
+                        return getInputModelForExtensibleEnum(m, values);
+                    };
```

Comment:
What happens here in `else` case?

Reply:
@AlexanderSher I didn't put a `break;` here. So this case will fall back to `default:` which will just return a string type.